### DISCUSSION TITLE
The purpose of the data store

### DISCUSSION THREAD

**Discussion by bill-anderson** [IATI/standard-process#7] [Ideas]
Historically various versions of the datastore have stuck to the argument that "IATI is not a database" and the datastore is therefore a searchable repository collating publishers data as is. The OECD CRS on the other hand is a (sort of) curated database. For that reason it is regarded as more reliable.

This could change if the datastore (or an augmented version) had the purpose of providing users with the most usable version of the data - transformed according to unambiguous and transparent rules.

This is relevant to discussions relating to both validation and changes to the standard.

**Comment by sylvanr** [comment]
In addition to an augmented datastore, we could consider a secretariat hosted, but open source and community developed "augmentation module", where customisations can be added for various different purposes (think of, for example currency conversion, date quarters, unified start and end dates, aggregated values for the different value subtypes). These additions would be niche, but could reduce the need for custom data stores. It'd be good to have some community outreach to see how many custom IATI data processors are actually currently in use.

**Comment by sylvanr** [comment]
On the subject of "IATI is not a database", this maybe Naive, but why not? 

It could also be considered to provide access to IATI data in multiple formats. So not only as a document store, but also as a Relational Database, NoSQL Database, a Graph Database, or as flattened tabular data (csv, xlsx, json).

This way, access to IATI data can be much more varied, and could attract outside users to start connecting to IATI data, and improve the overall usability and reach of IATI as a standard for humanitarian aid data.

**Reply by Melina-Loewer** [reply]
These thoughts sound interesting. However, I wonder if this wouldn't lead to an overwhelming amount of data, given that IATI data, unlike the annual and retrospective CRS data, consists of "current management data" that can change daily and therefore would need to be historicized in a database every day, right?

**Reply by HermanvanLoon** [reply]
I.m.o. IATI is a common, machine readable, language to exchange information about development and humanitarian aid in a structured way. Given the publication of raw IATI data, you can of course make a curated database tailored to specific user needs. The challenge is that there are many curation rules depending on the specific use-case. E.g. what exchange rate should one use when valuing amounts against a given reporting currency? How should traceability be represented, e.g. do you want attribution of financial amounts to the ultimate funders and if so, which methodology do you use? 

This still doesn't answer the question which was raised by @sylvanr about the technical standard to present the data (json, Graph database, etc.) which is also an important consideration. 

The comparison with the OECD/DAC database is legitimate. In addition to the comment of @Melina-Loewer, OECD/DAC reporting is a very centralized, manually curated process, with only a limited number of stakeholders, no results, documents and location data. Nonetheless, it still requires quite a substantial staff. 

To transform the IATI raw data in a curated database fit for specific use cases is far more complex than the OECD/DAC data, given the many publishers (1700+) and the broad scope of the IATI-standard. Careful consideration should be given if and how far we want to go to make this possible. E.g. can we define common curation rules for defining e.g. currency rate calculations and country attributions which would satisfy the needs of 90% of the data-users? Can we define a technical representation which would cover the needs of 90% of the data-users? These questions should i.m.o. be answered first before substantial effort is done to extend the current data-store platform.

**Reply by davidmegginson** [reply]
I agree with what Herman wrote, and also, how do we weight the users?  For example, even if partner-country governments counted for only 1% of IATI users, shouldn't their needs get a much bigger weight in our planning?

OECD CRS+ has the advantage that it's designed for a single type of user (OECD governments) and a single purpose (tracking aid spending retroactively).  We could build any number of single-purpose IATI-derived databases like that, and every one would be different.  At the Centre for Humanitarian Data, we did build one for tracking COVID spending and commitments, and it was successful (as far as it went), but still only a narrow snapshot of IATI taken from a very-specific point of view.

**Comment by siemvaessen** [comment]
The purpose for the [datastore](https://github.com/zimmerman-team/IATI.cloud) managed by Zimmerman has mainly been to accommodate user interfaces with IATI in different settings (use case depending) and a search to provide use cases with direct access to activities, transactions etc. The datastore managed by IATI is built around the same piece of technology: [Apache Solr.](https://solr.apache.org/) I did suggest to do a brief analysis to see what's necessary to merge those 2 datastores into a single one and to deprecate the older datastore and see what's needed from the old datastore that could potentially be migrated to the ONE Datastore. Not much chatter on that yet.

**Comment by davidmegginson** [comment]
**Why "IATI is not a database"?**

That was a starting requirement back in 2010.  I don't remember exactly why it existed, but it might have been that we didn't want to compete with the vendors and open projects who we (naïvely) believed at the time would be rushing in. 😕

**Purpose of the datastore**

We've been struggling with this same question with our Humanitarian API (HAPI) at the UN Centre for Humanitarian Data.  On the one hand, people want a data API to be as flexible as possible (hence, probably, the choice to use Solr for the IATI data stores); on the other hand, API users expect the data to be internally consistent and interoperable (hence our choice at the Centre to use PostgreSQL).  It's surprisingly hard to get uptake for "build your own bear" style data APIs — the power users say they want it, but they don't really need it, while the regular users don't have the skills or time to do all the extra work to make the data usable.
